docs/glossary.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Glossary
+authors:
+  - "@dhellmann"
+reviewers:
+  - "@timothysc"
+creation-date: 2019-06-10
+last-updated: 2019-06-10
+---
+
+<!--
+
+  Formatting instructions for this document:
+
+  - Create level 3 section for each glossary entry so other documents can consistently link to the definition.
+  - Use a bullet list to define sub-entries or variations.
+  - Wrap each sub-entry or variation in double underscores to have it rendered with emphasis.
+  - Use semantic linefeeds (https://rhodesmill.org/brandon/2012/one-sentence-per-line/)
+
+-->
+
+# Glossary
+
+This document defines the terms used in other documents created by the Cluster API group.
+
+## Table of Contents
+
+[A](#a) | [B](#b) | [C](#c) | [D](#d) | [H](#h) | [I](#i) | [K](#k) | [M](#m) | [O](#o) | [P](#p) | [S](#s) | [W](#w)
+
+## A
+
+### Add-ons
+
+Services beyond the fundamental components of Kubernetes.
+
+* __Core Add-ons__: Addons that are required to deploy a Kubernetes-conformant cluster: DNS, kube-proxy, CNI.
+* __Additional Add-ons__: Addons that are not required for a Kubernetes-conformant cluster (e.g. metrics/Heapster, Dashboard).
+
+## B
+
+### Bootstrap
+
+The process of turning a server into a Kubernetes node. This may involve assembling data to provide when creating the server that backs the Machine, as well as runtime configuration of the software running on that server.
+
+## C
+
+### Cluster
+
+A full Kubernetes deployment. See Management Cluster and Workload Cluster
+
+### Cluster API
+
+Or __Cluster API project__
+
+The Cluster API sub-project of the SIG-cluster-lifecycle. It is also used to refer to the software components, APIs, and community that produce them.
+
+### Control plane
+
+The set of Kubernetes services that form the basis of a cluster. See also https://kubernetes.io/docs/concepts/#kubernetes-control-plane There are two variants:
+
+* __Self-provisioned__: A Kubernetes control plane consisting of pods or machines wholly managed by a single Cluster API deployment.
+* __External__: A control plane offered and controlled by some system other than Cluster API (e.g., GKE, AKS, EKS, IKS).
+
+## D
+
+### Default implementation
+
+A feature implementation offered as part of the Cluster API project, infrastructure providers can swap it out for a different one.
+
+## H
+
+### Horizontal Scaling
+
+TBD
```

Comment:
The ability to add more machines based on policy and well defined metrics.  For example, add a machine to a cluster when CPU load average > (X) for a period of time (Y).

Reply:
should have added this earlier.

Reply:
How about if we leave adding that to another PR? I'd like everyone to get in the habit of updating this document as we go along and write other docs.

Reply:
The trick is to determine who will submit the PR and when? For non-trivial changes, I would recommend opening an issue. For trivial changes, it seems easier just to make the change and save the time of an additional lgtm/approve cycle.

Reply:
FWIW, I'd be willing to make this change today. Along with adding a link to the glossary to the GitBook...

Reply:
I'm still working out the process myself, so if you want me to make the changes here, that's fine. But I also don't mind filing a separate PR, or having you do it.